packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -431,6 +433,35 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
     throw new Error(`Invalid principal type for Lambda permission statement: ${principal.constructor.name}. ` +
       'Supported: AccountPrincipal, ArnPrincipal, ServicePrincipal');
   }
+
+  private parsePermissionPrincipalConditions(principal: iam.IPrincipal): { sourceAccount: string, sourceArn: string } | null {
```

Comment:
minor: massive function name. suggest something terser. It's a private function after all. Even just `parseConditions` will work.